packages/dev/core/src/FlowGraph/Blocks/Data/Math/flowGraphMatrixBlocks.ts

```diff
@@ -0,0 +1,255 @@
+import { Matrix, Quaternion, Vector3 } from "../../../../Maths/math.vector";
+import { FlowGraphBinaryOperationBlock } from "../flowGraphBinaryOperationBlock";
+import { RichTypeMatrix, RichTypeNumber, RichTypeQuaternion, RichTypeVector3 } from "../../../flowGraphRichTypes";
+import { FlowGraphUnaryOperationBlock } from "../flowGraphUnaryOperationBlock";
+import { FlowGraphBlock } from "../../../flowGraphBlock";
+import type { FlowGraphDataConnection } from "../../../flowGraphDataConnection";
+import type { FlowGraphContext } from "../../../flowGraphContext";
+
+/**
+ * Adds two matrices together.
+ * @experimental
+ */
+export class FlowGraphAddMatrixBlock extends FlowGraphBinaryOperationBlock<Matrix, Matrix, Matrix> {
+    constructor() {
+        super(RichTypeMatrix, RichTypeMatrix, RichTypeMatrix, (left, right) => left.add(right));
+    }
+}
+
+/**
+ * Adds a matrix and a number together.
+ * @experimental
+ */
+export class FlowGraphAddMatrixAndNumberBlock extends FlowGraphBinaryOperationBlock<Matrix, number, Matrix> {
+    private _cachedArray: Float32Array = new Float32Array(16);
+    private _cachedMatrix: Matrix = Matrix.Zero();
+    constructor() {
+        super(RichTypeMatrix, RichTypeNumber, RichTypeMatrix, (left, right) => {
+            for (let i = 0; i < left.m.length; i++) {
+                this._cachedArray[i] = left.m[i] + right;
+            }
+            return Matrix.FromArrayToRef(this._cachedArray, 0, this._cachedMatrix);

```

Comment:
This will always return the same Matrix. which is not like the other nodes.

I like it as it is less GC intensive but I think we should thrive for consistency and outline the rules.